### PR TITLE
Better error message for Firestore in Datastore Mode

### PR DESCRIPTION
**Pull request by samtstern**
### Description

See #2271 and #1988

### Scenarios Tested

**Project in Native Mode**
```bash
$ firebase --project=fir-dumpster init firestore

# ...

=== Firestore Setup

Firestore Security Rules allow you to define how and when to allow
requests. You can keep these rules in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore Rules? firestore.rules


? File firestore.rules already exists. Do you want to overwrite
 it with the Firestore Rules from the Firebase Console? No

Firestore indexes allow you to perform complex queries while
maintaining performance that scales with the size of the result
set. You can keep index definitions in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore indexes? firestore.ind
exes.json

i  Writing configuration info to firebase.json...
i  Writing project information to .firebaserc...
i  Writing gitignore file to .gitignore...

✔  Firebase initialization complete!
```

**Project in Datastore Mode**
```bash
$ firebase --project=fir-ds-mode-test init firestore

# ...

=== Firestore Setup

Error: It looks like this project is using Cloud Datastore or Cloud Firestore in Datastore Mode. The Firebase CLI can only manage projects using Firestore in Native Mode. For more information, see: https://cloud.google.com/datastore/docs/firestore-or-datastore
```

**Project without Firestore at all**
```bash
$ firebase --project=totally-empty-project init firestore

# ...

=== Firestore Setup

Error: It looks like you haven't used Cloud Firestore in this project before. Go to https://console.firebase.google.com/project/totally-empty-project/database to create your Cloud Firestore database.

Having trouble? Try firebase [command] --help
```

### CODE REVIEW

File: src/test/init/features/firestore.spec.ts [hunk 1]
```diff
@@ -41,8 +47,6 @@ describe("firestore", () => {
     });
 
     it("should error when cloud resource location is not set", async () => {
-      checkApiStub.returns(true);
-
       const setup = { config: {}, projectId: "my-project-123" };
 
       expect(firestore.doSetup(setup, {})).to.eventually.be.rejectedWith(
```
Comment (bkendall) [inline src/test/init/features/firestore.spec.ts:52]: These lint warnings are actually legit. Please use `await` on the `eventually` statements to ensure they resolve.

File: src/test/init/features/firestore.spec.ts [hunk 2]
```diff
@@ -61,5 +65,17 @@ describe("firestore", () => {
...
       );
     });
+
+    it("should error when firestore is in the wrong mode", async () => {
+      checkApiStub.returns(true);
+      checkDbTypeStub.returns("CLOUD_DATASTORE_COMPATIBILITY");
+
+      const setup = { config: {}, projectId: "my-project-123" };
+
+      expect(firestore.doSetup(setup, {})).to.eventually.be.rejectedWith(
```
Comment (bkendall) [inline src/test/init/features/firestore.spec.ts:75]: This, too, needs to be `await`'d

File: src/init/features/firestore/index.ts [hunk 3]
```diff
@@ -1,24 +1,45 @@
+import * as api from "../../../api";
```
Comment (bkendall) [inline src/init/features/firestore/index.ts:1]: Unused - remove.

File: src/init/features/firestore/index.ts [hunk 4]
```diff
@@ -1,24 +1,45 @@
...
 export async function doSetup(setup: any, config: any): Promise<void> {
+  const firestoreUnusedError = new FirebaseError(
+    `It looks like you haven't used Cloud Firestore in this project before. Go to ${clc.bold.underline(
+      `https://console.firebase.google.com/project/${setup.projectId}/database`
+    )} to create your Cloud Firestore database.`,
+    { exit: 1 }
+  );
+
+  // First check if the Firestore API is enabled. If it's not they need to go set up
+  // Firestore in the console.
```
Comment (rachelsaunders) [inline src/init/features/firestore/index.ts:22]:
```suggestion
  // First check if the Firestore API is enabled. If it's not, then the developer needs
  // to go set up Firestore in the console.
```

File: src/init/features/firestore/index.ts [hunk 5]
```diff
@@ -1,24 +1,45 @@
...
+  logger.debug(`database_type: ${dbType}`);
+
+  if (!dbType) {
+    throw firestoreUnusedError;
+  } else if (dbType !== "CLOUD_FIRESTORE") {
     throw new FirebaseError(
-      `It looks like you haven't used Cloud Firestore in this project before. Go to ${clc.bold.underline(
-        `https://console.firebase.google.com/project/${setup.projectId}/database`
-      )} to create your Cloud Firestore database.`,
+      `It looks like this project is using Cloud Datastore or Cloud Firestore in Datastore Mode. The Firebase CLI can only manage projects using Firestore in Native Mode. For more information, see: https://cloud.google.com/datastore/docs/firestore-or-datastore`,
```
Comment (rachelsaunders) [inline src/init/features/firestore/index.ts:42]:
```suggestion
      `It looks like this project is using Cloud Datastore or Cloud Firestore in Datastore mode. The Firebase CLI can only manage projects using Cloud Firestore in Native mode. For more information, visit https://cloud.google.com/datastore/docs/firestore-or-datastore`,
```

File: src/test/init/features/firestore.spec.ts [hunk 6]
```diff
@@ -61,5 +65,17 @@ describe("firestore", () => {
...
+        FirebaseError,
+        "It looks like this project is using Cloud Datastore or Cloud Firestore in Datastore Mode."
```
Comment (rachelsaunders) [inline src/test/init/features/firestore.spec.ts:77]:
```suggestion
        "It looks like this project is using Cloud Datastore or Cloud Firestore in Datastore mode."
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,3 +6,4 @@
 - Improves handling of permission denied errors during API enablement.
 - Removes deep proxying of `functions.config()` inside the Functions emulator.
 - Updates App Distribution upload URL.
+- Fixes error message when `firebase init firestore` is used on a project with Firestore in Datastore Mode.
```
Comment (rachelsaunders) [inline CHANGELOG.md:9]:
```suggestion
- Fixes error message when `firebase init firestore` is used on a project with Cloud Firestore in Datastore mode.
```

forgot this one! sorry!

**Pull request merged**